.github/workflows/ci.yml

```diff
@@ -44,3 +44,39 @@ jobs:
 
     - name: Run Tests
       run: dotnet test test\OpenFeature.Tests\ --configuration Release --logger:"console;verbosity=detailed"
+
+  packaging:
+    needs:
+    - unit-tests-linux
+    - unit-tests-windows
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup .NET SDK
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: |
+          6.0.x
+          7.0.x
+
+    - name: Restore
+      run: dotnet restore
+
+    - name: Pack NuGet packages (CI versions)
+      if: startsWith(github.ref, 'refs/heads/')
+      run: dotnet pack --no-restore --version-suffix "ci.$(date -u +%Y%m%dT%H%M%S)+sha.${GITHUB_SHA:0:9}"
+
+    - name: Pack NuGet packages (PR versions)
+      if: startsWith(github.ref, 'refs/pull/')
+      run: dotnet pack --no-restore --version-suffix "pr.$(date -u +%Y%m%dT%H%M%S)+sha.${GITHUB_SHA:0:9}"
+
+    - name: Pack NuGet packages (Release versions)
+      if: startsWith(github.ref, 'refs/tags/v')
+      run: dotnet pack --no-restore
+
+    - name: Publish NuGet packages
+      if: startsWith(github.ref, 'refs/heads/') || startsWith(github.ref, 'refs/pull/') || startsWith(github.ref, 'refs/tags/v')
+      run: dotnet nuget push "src/**/*.nupkg" --api-key "${GITHUB_TOKEN}" --source https://nuget.pkg.github.com/open-feature/index.json
```

Comment:
You'll need to use `secrets.NUGET_TOKEN`.

Reply:
Hmmm, I use `GITHUB_TOKEN` directly in my private repos, is the custom token related to public repo/forks stuff?

Reply:
In any event, updated and it seems happier so that's that!